converter.go

```diff
@@ -239,11 +240,21 @@ func valueToString(v driver.Value, tsmode snowflakeType, params map[string]*stri
 	logger.Debugf("TYPE: %v, %v", reflect.TypeOf(v), reflect.ValueOf(v))
 	if v == nil {
 		if tsmode == objectType || tsmode == arrayType || tsmode == sliceType {
-			return bindingValue{nil, "json", nil}, nil
+			return bindingValue{nil, jsonFormatStr, nil}, nil
 		}
 		return bindingValue{nil, "", nil}, nil
 	}
 	v1 := reflect.Indirect(reflect.ValueOf(v))
+
+	if valuer, ok := v.(driver.Valuer); ok { // check for driver.Valuer satisfaction and honor that first
+		if value, err := valuer.Value(); err == nil && value != nil {
+			// if the output value is a valid string, return that
+			if strVal, ok := value.(string); ok {
+				return bindingValue{&strVal, "", nil}, nil
+			}
```

Comment:
Leaving this here as a suggestion; did we want to let a marshaler handle the driver's value if not a string by default?

```suggestion
			} else if marshaled, err := json.Marshal(value); err == nil {
			    value := string(marshaled)
			    return bindingValue{&value, jsonFormatStr, nil}, nil
			}
```

Reply:
I'm not sure about that. This is only a writing part, what about reading into same type? I don't think we support such approach at all.

Reply:
Ok will close  this suggestion out.